### PR TITLE
Added VIA support for Keychron Q1V2

### PR DESCRIPTION
**Pull request by lalalademaxiya1**
Added VIA support for Keychron Q1v2 and request for reviews, Thanks!

## Description

- Added VIA JSONs for Keychron Q1V2.
  - Added the mcu name as the suffix to distinguish one from another.
  - Added ANSI, ISO and JIS versions including encoder support version each other.
  - Added JSONs to src and v3.

## QMK Pull Request 

Here is QMK PR(qmk/qmk_firmware#19307) but it may take a long time to be merged into qmk master.

## Checklist

- [ ] The VIA support for this keyboard is **MERGED** in QMK master already **(MANDATORY)**
- [x] The VIA definition follows the guide here: https://caniusevia.com/docs/layouts
- [x] I have a V3 JSON version for this keyboard definition.**(MANDATORY)**
- [x] I have tested this keyboard definition using VIA's "Design" tab.
- [x] I have tested this keyboard definition with firmware on a device.
- [x] I have assigned alpha keys and modifier keys with the correct colors.
- [x] The Vendor ID is not `0xFEED`

### CODE REVIEW

**Comment by wilba** [comment]
Please update the `"name": "Keychron Q1",` etc. for these definitions so they are unique.

For example,  `"name": "Keychron Q1V2 ANSI Knob"`

**Comment by lalalademaxiya1** [comment]
> 

Changed, Thanks for review again!

File: src/keychron/q1/ansi_stm32l432.json [hunk 1]
```diff
@@ -0,0 +1,291 @@
+{
+  "name": "Keychron Q1",
```
Comment (wilba) [inline src/keychron/q1/ansi_stm32l432.json:2]: Change to "Keychron Q1V2 ANSI"

File: src/keychron/q1/ansi_stm32l432_encoder.json [hunk 2]
```diff
@@ -0,0 +1,291 @@
+{
+  "name": "Keychron Q1",
```
Comment (wilba) [inline src/keychron/q1/ansi_stm32l432_encoder.json:2]: Change to "Keychron Q1V2 ANSI Knob"

File: src/keychron/q1/iso_stm32l432.json [hunk 3]
```diff
@@ -0,0 +1,294 @@
+{
+  "name": "Keychron Q1",
```
Comment (wilba) [inline src/keychron/q1/iso_stm32l432.json:2]: Change to "Keychron Q1V2 ISO"

**Review by wilba — CHANGES_REQUESTED** [review]
Make changes as suggested for ALL the definitions in this PR, including the ones being renamed/moved. Every keyboard name must be unique.

File: src/keychron/q1/ansi_stm32l432.json (hunk 1 above)
Comment (lalalademaxiya1) [reply src/keychron/q1/ansi_stm32l432.json]: changed.

File: src/keychron/q1/ansi_stm32l432_encoder.json (hunk 2 above)
Comment (lalalademaxiya1) [reply src/keychron/q1/ansi_stm32l432_encoder.json]: changed.

File: src/keychron/q1/iso_stm32l432.json (hunk 3 above)
Comment (lalalademaxiya1) [reply src/keychron/q1/iso_stm32l432.json]: changed.

**Comment by wilba** [comment]
Thanks. Please change all the other Keychron definitions in the same way, so that the `name` matches the layout and knob/no knob.

This will help users (and VIA team) when users install the wrong firmware, or have problems with VIA. We can tell what keyboard it is from the name in VIA.

**Pull request merged**